src/modules/scraper/adapter/gas-fees/gas-fees-service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from "@nestjs/common";
+import BigNumber from "bignumber.js";
+
+import { fixedPointAdjustment } from "../..//utils";
+import { EthProvidersService } from "../../../web3/services/EthProvidersService";
+import { chainIdToInfo } from "../../../../utils";
+import { MarketPriceService } from "../../../market-price/services/service";
+
+@Injectable()
+export class GasFeesService {
+  constructor(private providers: EthProvidersService, private marketPriceService: MarketPriceService) {}
+
+  public async getFillTxNetworkFee(destinationChainId: number, fillTxHash: string) {
+    const destinationChainProvider = this.providers.getProvider(destinationChainId);
+    const destinationChainInfo = chainIdToInfo[destinationChainId];
+
+    if (!destinationChainProvider || !destinationChainInfo) {
+      return {
+        fillTxBlockNumber: 0,
+        fee: "0",
+        feeUsd: "0",
+      };
+    }
+
+    const fillTxReceipt = await destinationChainProvider.getTransactionReceipt(fillTxHash);
+    // Some chains, e.g. Optimism, do not return the effective gas price in the receipt. We need to fetch it separately.
+    const gasPrice = fillTxReceipt.effectiveGasPrice || (await destinationChainProvider.getGasPrice());
```

Comment:
But the gas price fetched separately will be different than the gas price at tx time, right? Is this ok?

Reply:
I think we can actually remove that fallback. I looked at the docs again of ethers and the library will actually use the normal `gasPrice` if there is no concept of a `baseFee`. See https://docs.ethers.org/v5/api/providers/types/#providers-TransactionReceipt